mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/MassIndexingMonitor.java

```diff
@@ -21,9 +21,21 @@
  */
 public interface MassIndexingMonitor {
 
+	/**
+	 * Creates a type-group-specific monitor.
+	 * <p>
+	 * The mass indexer may group some of the types it has to index or index them separately.
+	 * The type group represents this combination of types that are retrieved for indexing
+	 * in the same pipeline.
+	 * When a particular type group is about to get indexed the mass indexer will request to create a type group monitor.
+	 *
+	 * @param context Describes the type group for which the monitor is requested.
+	 * @return A type group mass indexing monitor. By default, a no-op monitor is returned.
+	 * @see MassIndexingTypeGroupMonitor
+	 */
 	@Incubating
-	default MassIndexingTypeGroupMonitor typeGroupMonitor(MassIndexingTypeGroupMonitorContext context) {
-		return NoOpMassIndexingTypeGroupMonitor.INSTANCE;
+	default MassIndexingTypeGroupMonitor typeGroupMonitor(MassIndexingTypeGroupMonitorCreateContext context) {
+		return new LegacyDelegatingMassIndexingTypeGroupMonitor( this, context );
 	}
```

Comment:
I did not add the initialize method to `MassIndexingTypeGroupMonitor` since that'll be called in the exact same place where we call this one to create a type group, so users can do whatever they wanted in init in the constructor of the type group monitor.